src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,873 @@
+		//}
+
+		//void ViewModel.RemoveAt(uint32_t index)
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	var current = inner.GetAt(index).as< TreeViewNode > ();
+		//	inner.RemoveAt(index);
+
+		//	// Unhook event handlers
+		//	var tvnCurrent = get_self<TreeViewNode>(current);
+		//	tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector[index]);
+		//	tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector[index]);
+
+		//	// Remove tokens from vectors
+		//	m_collectionChangedEventTokenVector.erase(m_collectionChangedEventTokenVector.begin() + index);
+		//	m_IsExpandedChangedEventTokenVector.erase(m_IsExpandedChangedEventTokenVector.begin() + index);
+		//}
+
+		private void Append(object value)
+		{
+			//	GetVectorInnerImpl().Append(value);
+			//	TreeViewNode newNode = value.as< TreeViewNode > ();
+
+			//	// Hook up events and save tokens
+			//	var tvnNewNode = get_self<TreeViewNode>(newNode);
+			//	m_collectionChangedEventTokenVector.push_back(tvnNewNode.ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged }));
+			//	m_IsExpandedChangedEventTokenVector.push_back(tvnNewNode.AddExpandedChanged({ this, &ViewModel.TreeViewNodePropertyChanged }));
+			throw new NotImplementedException();
+		}
+
+		//void ViewModel.RemoveAtEnd()
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	var current = inner.GetAt(Size() - 1).as< TreeViewNode > ();
+		//	inner.RemoveAtEnd();
+
+		//	// unhook events
+		//	var tvnCurrent = get_self<TreeViewNode>(current);
+		//	tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector.back());
+		//	tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector.back());
+
+		//	// remove tokens
+		//	m_collectionChangedEventTokenVector.pop_back();
+		//	m_IsExpandedChangedEventTokenVector.pop_back();
+		//}
+
+		private void Clear()
+		{
+			//	// Don't call GetVectorInnerImpl().Clear() directly because we need to remove hooked events
+			//	unsigned int count = Size();
+			//	while (count != 0)
+			//	{
+			//		RemoveAtEnd();
+			//		count--;
+			//	}
+		}
+
+		//void ViewModel.ReplaceAll(array_view<IInspectable const> items)
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	return inner.ReplaceAll(items);
+		//}
+
+		//// Helper function
+		internal void PrepareView(TreeViewNode originNode)
+		{
+			// Remove any existing RootNode events/children
+			var existingOriginNode = m_originNode;
+			if (existingOriginNode != null)
+			{
+				for (int i = (existingOriginNode.Children.Count - 1); i >= 0; i--)
+				{
+					var removeNode = existingOriginNode.Children[i];
+					RemoveNodeAndDescendantsFromView(removeNode);
+				}
+
+				if (m_rootNodeChildrenChangedEventToken.value != 0)
+				{
+					existingOriginNode.Children.VectorChanged(m_rootNodeChildrenChangedEventToken);
+				}
+			}
+
+			//	// Add new RootNode & children
+			m_originNode = originNode;
+			m_rootNodeChildrenChangedEventToken = (originNode).ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)